contracts/registry/src/registry.rs

```diff
@@ -362,95 +362,110 @@ impl SBTRegistry for Contract {
     fn sbt_revoke_by_owner(&mut self, owner: AccountId, burn: bool) -> bool {
         let issuer = env::predecessor_account_id();
         let issuer_id = self.assert_issuer(&issuer);
-        let mut tokens_by_owner = self.sbt_tokens_by_owner(
-            owner.clone(),
-            Some(issuer.clone()),
-            None,
-            Some(MAX_REVOKE_PER_CALL),
-            Some(true),
-        );
-        let remaining_supply = self.sbt_supply_by_owner(owner.clone(), issuer.clone(), None);
-        let tokens_len = tokens_by_owner[0].1.len();
+        let tokens_by_owner =
+            self.sbt_token_ids_by_owner(owner.clone(), issuer_id, MAX_REVOKE_PER_CALL);
+
         if tokens_by_owner.is_empty() {
             return true;
-        };
-        let (_, tokens) = tokens_by_owner.pop().unwrap();
+        }
 
-        let mut token_ids = Vec::new();
+        let non_expired_count =
+            self.sbt_tokens_by_owner(owner.clone(), Some(issuer.clone()), None, None, Some(false))
+                [0]
+            .1
+            .len();
```

Comment:
so the extra call to `sbt_tokens_by_owner` is also needed in the previous implementation. It was missed there and there is a bug where when not burning tokens just changing their expire time in some cases it would never return true (the revocation process might never end). For the second part, the `sbt_tokens_by_owner` is unbounded. There is a limit within the method to limit the amount of tokens returned to avoid running out of gas